src/persist-client/src/internal/compact.rs

```diff
@@ -118,29 +119,35 @@ where
 
                     // Compaction is cpu intensive, so be polite and spawn it on the CPU heavy runtime.
                     let compact_span = debug_span!("compact::consolidate");
-                    let res = cpu_heavy_runtime
-                        .spawn_named(
-                            || "persist::compact::consolidate",
-                            Self::compact(
-                                cfg.clone(),
-                                Arc::clone(&blob),
-                                Arc::clone(&metrics),
-                                Arc::clone(&cpu_heavy_runtime),
-                                req,
-                                writer_id,
+                    let res = tokio::time::timeout(
+                        Duration::from_secs(60 * 10),
```

Comment:
is it crazy to do something like `max(60s, 1s per MiB)`? that would hopefully be something we could keep forever

either way, let's put any constants here as things on PersistConfig (i'm basically always going to say this when I see a constant in persist code :)

Reply:
I like that! adding both the `max` and moving to cfg